File: FingertipsUploadService/ProfileData/TimePeriod.cs
namespace FingertipsUploadService.ProfileData
{
    public class TimePeriod
    {
        public const int Undefined = -1;

        public int Year = Undefined;
        public int Quarter = Undefined;
        public int Month = Undefined;
        public int YearRange = Undefined;

        public bool IsQuarterly
        {
            get { return Quarter > 0; }
        }

        public bool IsMonthly
        {
            get { return Month > 0; }
        }

        public bool IsLaterThan(TimePeriod period)
        {
            if (Year > period.Year)
            {
                return true;
            }

            if (Year < period.Year)
            {
                return false;
            }

            if (IsQuarterly)
            {
                return Quarter > period.Quarter;
            }

            if (IsMonthly)
            {
                return Month > period.Month;
            }

            return false;
        }
    }
}

File: IndicatorsUI/MainUI/angular-app/src/app/shared/classes/time-period.ts
import { Grouping } from '../../typings/FT';
import { String } from 'typescript-string-operations';

export class TimePeriod {
    year: number;
    yearRange: number;
    quarter: number;
    month: number;

    constructor(private grouping: Grouping) {
        this.year = this.grouping.DataPointYear;
        this.yearRange = this.grouping.YearRange;
        this.quarter = this.grouping.DataPointQuarter;
        this.month = this.grouping.DataPointMonth;
    }

    getSortableNumber(): string {
        const quarter = this.quarter === -1 ? 0 : this.quarter;
        const month = this.month === -1 ? 0 : this.month;
        return String.Format('{0:0000}{1:00}{2:00}', this.year, quarter, month);
    }
}
